### PR TITLE
feat: tab component

### PR DESCRIPTION
**Pull request by Marine-Berthier**
References [VSST-479](https://autoricardo.atlassian.net/browse/VSST-479?atlOrigin=eyJpIjoiYjU4ZTRkZTFhNDQ3NDAzYzk3NDdhZjE3YTNjMzg5NjUiLCJwIjoiaiJ9)

## Motivation and context

For the first step of the insertion process we need to use Tabs
[Here is the design](https://www.figma.com/file/CHU4fpMfXpiAW4qNFzi9y1/11.-AS24---Web---Components?node-id=1862%3A59673&t=ButXzNCT5VJmRiQg-0)

## Before

No tab components

## After

Tab component

## How to test

Please make sure that the component works as intended and matches the design, [here](https://marine-vsst-479-components-pkg.branch.autoscout24.dev/?path=/story/components-navigation-tabs--overview).
Note: the selected border bottom can be used as provided by chakra: 
dc0d6fe4-d633-451f-be85-d58cca4b6fbb

Thank you

[VSST-479]: https://autoricardo.atlassian.net/browse/VSST-479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[VSST-479]: https://autoricardo.atlassian.net/browse/VSST-479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/hide/index.stories.mdx [hunk 1]
```diff
@@ -111,6 +111,7 @@ If you are using testing-library, you will face some limitations in testing diff
 ```tsx
 /* eslint-disable import/no-unresolved, unicorn/filename-case */
 import { render, screen, within } from '@testing-library/react';
+
```
Comment (Marine-Berthier) [inline src/components/hide/index.stories.mdx:114]: This is a fix from the linter

File: src/components/show/index.stories.mdx [hunk 2]
```diff
@@ -111,6 +111,7 @@ If you are using testing-library, you will face some limitations in testing diff
 ```tsx
 /* eslint-disable import/no-unresolved, unicorn/filename-case */
 import { render, screen, within } from '@testing-library/react';
+
```
Comment (Marine-Berthier) [inline src/components/show/index.stories.mdx:114]: Also a linter fix

**Review by Milos-Zbiljic — APPROVED** [review]

File: src/themes/components/tabs.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  color: 'gray.600',
+  _selected: {
+    fontWeight: 'bold',
+    color: 'gray.900',
+    borderColor: 'currentColor',
+  },
+  _disabled: {
+    _active: { bg: 'none' },
+    cursor: 'not-allowed',
+    opacity: 0.4,
```
Comment (GutierrezGo) [inline src/themes/components/tabs.ts:18]: @Marine-Berthier you need to reference the opacity through the object we already have in place instead hardcoding the opacity.

File: src/themes/components/tabs.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const baseStyleTab = {
+  textStyle: 'body',
+  transitionProperty: 'common',
+  transitionDuration: 'normal',
+  borderBottom: '1.5px solid',
```
Comment (GutierrezGo) [inline src/themes/components/tabs.ts:5]: We don't have this token in place. Is it possible to check with the designer if we can use `1px` or `2px`? or if it makes sense for future uses introduce this new token.

Would you mind checking this with the designer? Thank you!

File: src/themes/components/tabs.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  _disabled: {
+    _active: { bg: 'none' },
+    cursor: 'not-allowed',
+    opacity: 0.4,
+  },
+};
+
+const baseStyleTablist = {
+  borderBottom: '1.5px solid',
```
Comment (GutierrezGo) [inline src/themes/components/tabs.ts:23]: We don't have this token in place. Is it possible to check with the designer if we can use 1px or 2px? or if it makes sense for future uses introduce this new token.

Would you mind checking this with the designer? Thank you!

File: src/components/tab/__tests__/Tabs.Test.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    renderWrapper({});
+    userEvent.click(screen.getByText('Two'));
+    expect(screen.getByText('Zwei')).toBeInTheDocument();
+  });
+  it('displays the tabs with the last tab opens by default', async () => {
+    renderWrapper({ defaultIndex: 2 });
+    expect(screen.getByText('Drei')).toBeInTheDocument();
+    expect(screen.queryByText('Eins')).not.toBeInTheDocument();
+  });
+  it('displays the tabs with one tab disabled', async () => {
```
Comment (GutierrezGo) [inline src/components/tab/__tests__/Tabs.Test.tsx:59]: I'm wondering the reason of using `async` in the tests. Would you mind giving more context here?
Thank you!

**Pull request merged**